graviton/blackbox.py

```diff
@@ -1309,7 +1353,11 @@ def extract_logs(cls, txn):
 
     @classmethod
     def extract_cost(cls, txn):
-        return txn.get("cost")
+        # cost is treated as a derived property if budget-consumed and budget-added is available
+        if "budget-consumed" in txn and "budget-added" in txn:
+            return txn["budget-consumed"] - txn["budget-added"]
```

Comment:
I'm trying to understand why cost should be the difference between `budget-consumed` and `budget-added` instead of just an alias for `budget-consumed` and therefore always positive. I don't have a strong preference here but if there is already this assumption elsewhere in our codebase, I would be convinced that we should make the change. In the case that no budget added (i.e., all the pre-existing test cases where cost was computed), this isn't an issue. Also, an argument for the change is if the previous cost when budget was added, was already negative (which I vaguely recall is the case).

Reply:
In my opinion, we normally think of cost over a certain period as being the difference between the original overall opcode budget and the current overall opcode budget, so that was what I was aiming for here.

> Also, an argument for the change is if the previous cost when budget was added, was already negative (which I vaguely recall is the case)

Cost on the dryrun side is also calculated the way I describe above. The problem is that it's a `uint` so it would just underflow instead of going negative.